src/features/class-open-graph.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Open_Graph class file
+ *
+ * @package wp-seo
+ */
+
+namespace Alley\WP\WP_SEO\Features;
+
+use Alley\WP\Types\Feature;
+
+use function Alley\WP\WP_SEO\register_meta_helper;
+
+/**
+ * Open Graph Feature
+ */
+final class Open_Graph implements Feature {
+	/**
+	 * Boot the feature.
+	 */
+	public function boot(): void {
+		add_action( 'init', [ $this, 'add_post_type_support' ] );
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_assets' ] );
+		add_action( 'init', [ $this, 'add_meta_fields' ] );
+		add_action( 'wp_head', [ $this, 'render_open_graph_tags' ] );
+	}
+
+	/**
+	 * Add post type support.
+	 *
+	 * @todo: Supported post types should be configurable from admin page.
+	 *
+	 * @return void
+	 */
+	public function add_post_type_support() {
+		add_post_type_support( 'post', 'open-graph' );
+		add_post_type_support( 'page', 'open-graph' );
+	}
+
+
+	/**
+	 * Enqueue block editor assets.
+	 *
+	 * @return void
+	 */
+	public function enqueue_block_editor_assets() {
+		global $post;
+
+		if ( ! is_admin()
+			|| ! ( $post instanceof \WP_Post || is_int( $post ) || is_null( $post ) )
+		) {
+			return;
+		}
+
+		$post_type = get_post_type( $post );
+
+		if ( empty( $post_type ) || ! post_type_supports( $post_type, 'open-graph' ) ) {
+			return;
+		}
+
+		wp_enqueue_script( 'wp-seo-open-graph-js' );
+	}
+
+	/**
+	 * Add meta fields.
+	 */
+	public function add_meta_fields(): void {
+		register_meta_helper(
+			'post',
+			get_post_types_by_support( 'open-graph' ),
+			'wp_seo_open_graph_title',
+			[
+				'sanitize_callback' => 'wp_kses_post',
```

Comment:
could probably use `sanitize_text_field` here. I don't think we allow/need any html, right?

Reply:
Good call. Updated in [14ef3a6](https://github.com/alleyinteractive/wp-seo/pull/132/commits/14ef3a6c11ab385d7c20d1184614698efe799f8f)